airframe-surface/shared/src/main/scala/wvlet/airframe/surface/SurfaceMacros.scala

```diff
@@ -571,6 +590,21 @@ private[surface] object SurfaceMacros {
         q"wvlet.airframe.surface.ExistentialType"
       case t @ RefinedType(List(_, baseType), decl) =>
         newGenericSurfaceOf(baseType)
+      case t if hasStringUnapply(t) =>
+        t.typeSymbol.fullName.split("\\.").toList match {
+          case p1 :: tl =>
+            // companion object package -> Select(Select( .... ))) tree
+            val ex = tl.foldLeft[Tree](Ident(TermName(p1))) { (z, n) =>
+              Select(z, TermName(n))
+            }
+            val expr = q"""wvlet.airframe.surface.EnumSurface(classOf[${finalTypeOf(t)}], { (cl: Class[_], s: String) =>
+                ${ex}.unapply(s).asInstanceOf[Option[Any]]
+              }
+            )"""
+            expr
+          case _ =>
+            newGenericSurfaceOf(t)
```

Comment:
This means we cannot define Enum-like classes in the default package? But in reality, we almost always define a package.

Reply:
OK. I'll add support for unapply methods in package objects.